pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/Connection.java

```diff
@@ -190,8 +192,10 @@ public Future<ResultSetGroup> executeAsync(@Nullable String tableName, String qu
   private static String[] resolveTableName(String query) {
     try {
       SqlNodeAndOptions sqlNodeAndOptions = CalciteSqlParser.compileToSqlNodeAndOptions(query);
-      List<String> tableNames = CalciteSqlParser.extractTableNamesFromNode(sqlNodeAndOptions.getSqlNode());
-      return tableNames.toArray(new String[0]);
+      String tableName = (sqlNodeAndOptions.getSqlNode() != null
```

Comment:
sqlNodeAndOptions.getSqlNode() will never be null, this check is not needed

Reply:
@walterddr updated this! 

On a side note, i wonder if we can remove the same logic from here as well - https://github.com/apache/pinot/blob/2b47dd68926dd39da05949fc204388797dfd14cb/pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java#L232